pkg/datapath/linux/node.go

```diff
@@ -851,6 +851,11 @@ func (n *linuxNodeHandler) insertNeighbor(ctx context.Context, newNode *nodeType
 	}
 }
 
+func (n *linuxNodeHandler) InsertMiscNeighbor(newNode *nodeTypes.Node) {
+	ctx := context.Background()
+	n.insertNeighbor(ctx, newNode, false)
+}
```

Comment:
What happens if this fails? `insertNeighbor` looks like it doesn't return any errors which is not ideal, but this leads to us building on top of code that can be quite fragile in case something fails.

I believe there might be some parallel work here by @derailed to improve this, but not sure of the status of it. I don't want to add the entire scope of that work into this PR, but I wanted to make sure there's at least a synchronization point here regarding that. If it's not feasible, then it's worth understanding the risks of building on top of code which doesn't handle errors properly and see what we can do about it.

Reply:
It's a good question, I think there are multiple aspects. Looking at upsertService() in failure cases it just returns with lb.ID(0) and an err, but I'm currently not seeing a cleanup of resources that were installed prior. We could do the same and just propagate, or have some sort of reconciler in general in case anything which affects datapath state in upsertService() errors. What are your thoughts?

Reply:
I've added the doc comment, and for the time being removed the nil err. Turns out there is currently no error I could propagate from the node manager in the first place.

Reply:
I've found the PR: https://github.com/cilium/cilium/pull/29415. Given that the conflicts might be non-trivial, I'd probably opt to not block this PR on that. 

> ignore deletion errors given we don't want to abort a service delete..), or have some sort of reconciler in general in case anything which affects datapath state in upsertService() errors. What are your thoughts?

I think ignoring errors is the main risk. Fernand's PR above moves us in the right direction in the short term, but long term, yes a reconciler (based on [general reconciler](https://github.com/cilium/cilium/pull/29698/), AFAIU) for the node manager / node neighbor manager would be ideal. 

However, for the sake of expediency, it might help to take a look at the code in a more practical way: If we ignore deletion errors, does that open us up to a risk of a stale neighbor entry potentially conflicting with another entry (maybe for another node in the case of IP reuse, or something like that)? If so, how likely is that risk and how could we mitigate it?

Reply:
> Looking at upsertService() in failure cases it just returns with lb.ID(0) and an err, but I'm currently not seeing a cleanup of resources that were installed prior.

Should we file a separate issue for that? Maybe a good-first-issue?

Reply:
> However, for the sake of expediency, it might help to take a look at the code in a more practical way: If we ignore deletion errors, does that open us up to the risk of a leaked, stale neighbor entry that potentially conflicts with another entry in the future (maybe with another node in the case of node IP reuse on node deletion/creation, or something like that)? If so, how likely is that risk and how could we mitigate it?

In this specific case it won't given we only tell the kernel "please resolve IP xyz" for us, and the kernel automatically keeps this up to date, so this is not like the old days where we added permanent entries (see also: https://isovalent.com/blog/post/2021-12-release-111/#managed-ipv4-ipv6-discovery ).

> Should we file a separate issue for that? Maybe a good-first-issue?

I'll open one, I don't think it's a good first issue, but we'll take care of it (e.g. Aditi or myself).

Reply:
Added these two:
- https://github.com/cilium/cilium/issues/31042
- https://github.com/cilium/cilium/issues/31043